### PR TITLE
Extracted destination section to its own component

### PR DESCRIPTION
**Pull request by pedroyan**
https://drive.google.com/file/d/1TH7-uKKCTKGVKnF-aSffGG5rkzBDKGVs/view?usp=sharing

### CODE REVIEW

**Comment by pedroyan** [comment]
@barryokane I've coded a mechanism to detect if the JS is running in the patternlib and if so, it will mock an API call for better visualization of the entire component

File: source/_patterns/03-organisms/referrer-destination-renderer.twig [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<input id="pattern_lib_mock_marker" type="hidden" value="true"/>
+
+{% include "molecules-referrer-destination-form" with {
```
Comment (barryokane) [inline source/_patterns/03-organisms/referrer-destination-renderer.twig]: Patternlib supports json files for test data (both globally and per pattern file). So I move the test model for this to a separate .json file.

**Review by barryokane — APPROVED** [review]
Nice work.

Patternlib supports separate json files for test data (both globally and for each specific pattern) - so I move your test data.  Otherwise all good.

**Comment by barryokane** [comment]
Wait - I'm getting JS errors...

**Review by barryokane — CHANGES_REQUESTED** [review]
I really like the idea of mocking the API call in the pattern.

However, as ui-scripts.js is copied exactly as is to the live site, that means we'll have a bunch of code that is not relevant to live.  And that file is complex/messy enough as it is (and it is not currently minified).

So, my preference would be to move the mock to a separate js file that is only used on the patternlib

File: source/js/ui-scripts.js [hunk 2]
```diff
@@ -72,239 +85,325 @@ function hideLoader() {
...
-	  $('.toggleCheckbox tr').click(function() {
-			  $(this).find('td input:checkbox').toggleCheckbox();
-	  })
-	  $('.toggleCheckbox tr td input:checkbox').click(function(event) {
-			  event.stopPropagation();
-	  })
-  });
+var groupMockHtml = " <style>#group_span fieldset{margin-top: 15px;}#group_span fieldset legend{width: auto; margin-bottom: 0; border-bottom: 0; font-size: 1.2em;}#group_span fieldset b{font-weight: normal;}</style> <fieldset> <legend> Automatic selection </legend> <b>One specialist will be selected automatically from:</b><br/> &nbsp&nbsp&nbsp&nbsp<b>Dr Roland Rat</b><br/> &nbsp&nbsp&nbsp&nbsp<b>Professor Wise Old Owl</b><br/> &nbsp&nbsp&nbsp&nbsp<b>Mr Fred Flintstone</b><br/> </fieldset>"
 
+function validateGroup(){
```
Comment (pedroyan) [inline source/js/ui-scripts.js:175]: Leave this function here, but don't modify it with mock stuff.

Instead, intercept the onClick function using jQuery and execute the mock when running JS inside the patternlib

**Pull request closed without merging**